### PR TITLE
Move settings classes to packages

### PR DESCRIPTION
**Pull request by NotWoods**
Cleans up the settings folder a little bit by moving classes to subfolders.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt [hunk 1]
```diff
@@ -44,7 +44,10 @@ class DeleteBrowsingDataFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        controller = DefaultDeleteBrowsingDataController(context!!)
+        controller =
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt]: nit: Not sure if we need to make this change.

File: app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt [hunk 2]
```diff
@@ -136,15 +139,17 @@ class DeleteBrowsingDataFragment : Fragment() {
         progress_bar.visibility = View.VISIBLE
         delete_browsing_data_wrapper.isEnabled = false
         delete_browsing_data_wrapper.isClickable = false
-        delete_browsing_data_wrapper.alpha = DISABLED_ALPHA
+        delete_browsing_data_wrapper.alpha =
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt]: nit: Not sure if we need to make this change.

File: app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt [hunk 3]
```diff
@@ -136,15 +139,17 @@ class DeleteBrowsingDataFragment : Fragment() {
...
+            DISABLED_ALPHA
     }
 
     private fun finishDeletion() {
         val popAfter = open_tabs_item.isChecked
         progress_bar.visibility = View.GONE
         delete_browsing_data_wrapper.isEnabled = true
         delete_browsing_data_wrapper.isClickable = true
-        delete_browsing_data_wrapper.alpha = ENABLED_ALPHA
+        delete_browsing_data_wrapper.alpha =
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt]: nit: Not sure if we need to make this change.

File: app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt [hunk 4]
```diff
@@ -28,7 +29,8 @@ class SearchEngineFragment : PreferenceFragmentCompat() {
                 isChecked = context.settings.shouldShowSearchSuggestions
             }
 
-        searchSuggestionsPreference?.onPreferenceChangeListener = SharedPreferenceUpdater()
+        searchSuggestionsPreference?.onPreferenceChangeListener =
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt]: nit: Not sure if we need to make this change.

File: app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt [hunk 5]
```diff
@@ -45,8 +47,11 @@ class SearchEngineFragment : PreferenceFragmentCompat() {
                 isChecked = context.settings.shouldShowClipboardSuggestions
             }
 
-        showHistorySuggestions?.onPreferenceChangeListener = SharedPreferenceUpdater()
-        showBookmarkSuggestions?.onPreferenceChangeListener = SharedPreferenceUpdater()
-        showClipboardSuggestions?.onPreferenceChangeListener = SharedPreferenceUpdater()
+        showHistorySuggestions?.onPreferenceChangeListener =
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt]: nit: Not sure if we need to make this change.

**Comment by sblatz** [comment]
@NotWoods could you rebase from master and fix merge conflicts? Then we can merge this :)

**Comment by sblatz** [comment]
@NotWoods looks like TaskCluster is having some issues with `DefaultBrowserToolbarControllerTest`: https://taskcluster-artifacts.net/Or4YtjM6QByviZRG0lcFBg/0/public/logs/live_backing.log

**Pull request merged**